ordered_product/tests.py

```diff
@@ -0,0 +1,147 @@
+import pytest
+from ordered_product.models import Supplier, Client, Product, SupplierProduct, DeliveryLocation, OrderedProduct
+import datetime
+from django.core.exceptions import ValidationError
+
+
+@pytest.fixture
+def supplier0():
+    return Supplier(user_name="bestSup",
+                    first_name="Yuval",
+                    last_name="Cohen",
+                    password="1234",
+                    business_name="bestSup")
+
+
+@pytest.fixture
+def client0():
+    return Client(user_name="meitar1996",
+                  first_name="Meitar",
+                  last_name="Rizner",
+                  password="123456",
+                  area="North Israel")
+
+
+@pytest.fixture
+def product0():
+    return Product(qr_code="Q5o76MbdiNbXprNEnHfpcGWFp1CMF8XY",
+                   product_name="Apple",
+                   description="Sweety!")
+
+
+@pytest.fixture
+def supplier_product0(product0, supplier0):
+    return SupplierProduct(suppler_product_id=1,
+                           qr_code=product0,
+                           user_name=supplier0,
+                           price=10,
+                           quantity=50)
+
+
+@pytest.fixture
+def delivery_location0(supplier0):
+    return DeliveryLocation(user_name=supplier0,
+                            location="Kiryat Shmona",
+                            date=datetime.date(2022, 12, 30))
+
+
+@pytest.fixture
+def ordered_product0(delivery_location0, client0, supplier_product0):
+    return OrderedProduct(
+        delivery_location_id=delivery_location0,
+        user_name=client0,
+        supplier_product_id=supplier_product0,
+        quantity=3)
+
+
+class TestOrderedProductModel():
+    @pytest.mark.django_db()
+    def test_save_and_delete_ordered_product(self, delivery_location0, supplier_product0, ordered_product0):
+        delivery_location0.save()
+        supplier_product0.save()
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 in OrderedProduct.objects.all()
+        ordered_product0.delete_ordered_product()
+        assert ordered_product0 not in OrderedProduct.objects.all()
+
+    @pytest.mark.django_db()
+    def test_filter_by_id(self, delivery_location0, supplier_product0, ordered_product0):
+        delivery_location0.save()
+        supplier_product0.save()
```

Comment:
All your tests start with those 2 lines. I would make sense to put that inside a fixture.